src/drive/web/modules/drive/Toolbar/components/useAppLinkWithStoreFallBack.jsx

```diff
@@ -0,0 +1,41 @@
+import { useEffect, useState } from 'react'
+import { Q, models } from 'cozy-client'
+
+const { applications } = models
+
+const useAppLinkWithStoreFallback = (slug, client, path = '') => {
+  const [fetchStatus, setFetchStatus] = useState('loading')
+  const [isInstalled, setIsInstalled] = useState(true)
+  const [url, setURL] = useState()
+
+  useEffect(
+    () => {
+      const load = async () => {
+        try {
+          const apps = await client.query(Q('io.cozy.apps'))
+          const appDocument = { slug }
+          const appInstalled = applications.isInstalled(apps.data, appDocument)
+          setIsInstalled(!!appInstalled)
+          if (appInstalled) {
+            setURL(applications.getUrl(appInstalled) + path)
+          } else {
+            setURL(applications.getStoreURL(apps.data, appDocument))
+          }
+          setFetchStatus('loaded')
+        } catch (error) {
+          setFetchStatus('errored')
+        }
+      }
+      load()
+    },
+    [client, slug, path]
+  )
+
+  return {
+    fetchStatus,
+    isInstalled,
+    url
+  }
+}
+
+export default useAppLinkWithStoreFallback
```

Comment:
Reminds me of something 🤓

Reply:
Let's me add it in Cozy-Client ;).

Reply:
https://github.com/cozy/cozy-client/pull/651